src/network.rs

```diff
@@ -327,26 +335,36 @@ fn _network_read_range(
     let end = offset as usize + size_to_read - 1;
     let hvalue = format!("bytes={offset}-{end}");
     let hd = unsafe { &mut *(handle as *const c_void as *mut HandleData) };
-    let clt = Client::builder().build()?;
-    let initial = clt.request(Method::GET, &hd.url);
-    let in_case_of_error = initial
-        .try_clone()
-        .ok_or(ProjError::RequestCloneError)?
-        .header("Range", &hvalue)
-        .header("Client", CLIENT);
-    let req = in_case_of_error
-        .try_clone()
-        .ok_or(ProjError::RequestCloneError)?;
-    let mut res = req.send()?;
+    let clt = Agent::new();
+    let initial = clt.get(&hd.url);
+    let in_case_of_error = initial.clone().set("Range", &hvalue).set("Client", CLIENT);
+    let req = in_case_of_error.clone();
+    let mut res = req.call()?;
     // hand the response and retry instance off to the error-handler, continue on success
     error_handler(&mut res, in_case_of_error)?;
-    let headers = res.headers().clone();
-    let contentlength = res.content_length().ok_or(ProjError::ContentLength)? as usize;
+    let headers = res
+        .headers_names()
+        .into_iter()
+        .filter_map(|h| {
+            Some({
+                let v = res.header(&h)?.to_string();
+                (h, v)
+            })
+        })
+        .collect();
+    let Some(Ok(contentlength)) = res.header("Content-Length").map(str::parse::<usize>) else {
+        return Err(ProjError::ContentLength);
+    };
     // Copy the downloaded bytes into the buffer so it can be passed around
+    let capacity = contentlength.min(size_to_read);
```

Comment:
This is a pre-existing issue, but we aren't consistent about `out_size_read`. If `Content-Length` is larger than `size_to_read`, we set `out_size_read` to a value larger than the actual length of the buffer. We should just set it to `buf.len()`, regardless or how we handle that case.

Same for `_network_open` of course.